### PR TITLE
Filipe/substreams option runtime

### PR DESCRIPTION
**Pull request by mangas**

### CODE REVIEW

**Comment by leoyvens** [comment]
Why isn't it sufficient to use a fake `RuntimeHostBuilder`?

**Comment by leoyvens** [comment]
Hmm I see that replacing the `RuntimeHostBuilder` isn't so simple.

**Comment by mangas** [comment]
> Hmm I see that replacing the `RuntimeHostBuilder` isn't so simple.

This way is also more explicit and requires less "hacks" IMO

File: core/src/subgraph/instance.rs [hunk 1]
```diff
@@ -90,7 +94,12 @@ where
...
         host_metrics: Arc<HostMetrics>,
-    ) -> Result<T::Host, Error> {
+    ) -> Result<Option<T::Host>, Error> {
+        let host_builder = match self.host_builder {
+            None => return Ok(None),
+            Some(ref host_builder) => host_builder,
+        };
+
         let mapping_request_sender = {
             let module_bytes = data_source.runtime();
```
Comment (leoyvens) [inline core/src/subgraph/instance.rs:104]: What about making this controlled by the data source, so we change `DataSource::runtime` to return an option, and if it returns none then no host is created.

**Review by leoyvens — CHANGES_REQUESTED** [review]

File: runtime/wasm/src/host.rs [hunk 2]
```diff
@@ -53,7 +53,7 @@ impl<C: Blockchain> RuntimeHostBuilderTrait<C> for RuntimeHostBuilder<C> {
     type Req = MappingRequest<C>;
 
     fn spawn_mapping(
-        raw_module: Vec<u8>,
+        raw_module: &Vec<u8>,
```
Comment (leoyvens) [inline runtime/wasm/src/host.rs:56]:
```suggestion
        raw_module: &[u8],
```

File: runtime/wasm/src/mapping.rs [hunk 3]
```diff
@@ -13,15 +13,15 @@ use std::thread;
 
 /// Spawn a wasm module in its own thread.
 pub fn spawn_module<C: Blockchain>(
-    raw_module: Vec<u8>,
+    raw_module: &Vec<u8>,
```
Comment (leoyvens) [inline runtime/wasm/src/mapping.rs:16]:
```suggestion
    raw_module: &[u8],
```

**Review by leoyvens — CHANGES_REQUESTED** [review]

**Review by leoyvens — APPROVED** [review]

**Pull request merged**